File: backend/app/Models/StudentAssignment.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class StudentAssignment extends Model
{
    use HasFactory;

    public $timestamps = false;

    protected $fillable = [
        "assignment_id",
        "student_task_name" ,
        "student_task",
        "user_id",
        "grade",
    ];

    public function assignment() : BelongsTo
    {
        return $this->belongsTo(Assignment::class);
    }

    public function user() : BelongsTo
    {
        return $this->belongsTo(User::class);
    }
}

File: tests/BolcsessegBarkaAPITests/ApiLibrary/StudentAssignment.cs
using Newtonsoft.Json;

namespace ApiLibrary;

public class StudentAssignment
{
    [JsonProperty("id")]
    
    public int Id { get; set; }
    
    [JsonProperty("assignment_id")]
    
    public int AssignmentId { get; set; }
    
    [JsonProperty("student_task_name")]
    
    public string? StudentTaskName { get; set; }
    
    [JsonProperty("student_task")]
    
    public string StudentTask { get; set; }
    
    [JsonProperty("user_id")]
    
    public int? UserId { get; set; }
}